lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1297,12 +1297,10 @@ public function validateIdentifier()
     public function validateAssociations()
     {
         foreach ($this->associationMappings as $mapping) {
-            if (
-                ! class_exists($mapping['targetEntity'])
-                && ! interface_exists($mapping['targetEntity'])
-                && ! trait_exists($mapping['targetEntity'])
-            ) {
-                throw MappingException::invalidTargetEntityClass($mapping['targetEntity'], $this->name, $mapping['fieldName']);
+            /** @var string */
```

Comment:
Without this, I'd have to suppress `TypeDoesNotContainType` for the entire `ClassMetadataInfo`. 

Given this class benefits a lot form static analysis, I'd really like to avoid having to do that. I'm open for alternative suggestions how that could be achieved.